configuration/pih/reports/sql/covid19encounterexport.sql

```diff
@@ -6,98 +6,1328 @@
+                    AND encounter_id IN (SELECT 
+                        encounter_id
+                    FROM
+                        encounter
+                    WHERE
+                        encounter_type = @covid_admission_encounter_type))
+            AND concept_id = CONCEPT_FROM_MAPPING('PIH', 'Radiology procedure performed')
+            AND voided = 0) o1 ON e.patient_id = o1.person_id
+        AND e.encounter_id = o1.encounter_id 
+SET 
+    radiology_other = CONCEPT_NAME(o1.value_coded, 'fr');
+
+UPDATE temp_encounter e
+        LEFT JOIN
+    (SELECT 
+        encounter_id, person_id, obs_group_id, value_text
+    FROM
+        obs
+    WHERE
+        obs_group_id IN (SELECT 
+                obs_group_id
+            FROM
+                obs o
+            WHERE
+                voided = 0
+                    AND concept_id = CONCEPT_FROM_MAPPING('PIH', 'Radiology procedure performed')
+                    AND value_coded NOT IN (CONCEPT_FROM_MAPPING('CIEL', '165152') , CONCEPT_FROM_MAPPING('CIEL', '163041'), CONCEPT_FROM_MAPPING('CIEL', '845'))
+                    AND encounter_id IN (SELECT 
+                        encounter_id
+                    FROM
+                        encounter
+                    WHERE
+                        encounter_type = @covid_admission_encounter_type))
+            AND concept_id = CONCEPT_FROM_MAPPING('PIH', 'Radiology report comments')
+            AND voided = 0) o1 ON e.patient_id = o1.person_id
+        AND e.encounter_id = o1.encounter_id 
+SET 
+    radiology_other_comments = o1.value_text;
+
+UPDATE temp_encounter 
+SET 
+    disposition = OBS_VALUE_CODED_LIST(encounter_id,
+            'PIH',
+            'Hum Disposition categories',
+            'fr');
+UPDATE temp_encounter 
+SET 
+    admission_ward = (SELECT 
+            name
+        FROM
+            location
+        WHERE
+            location_id = OBS_VALUE_TEXT(encounter_id,
+                    'PIH',
+                    'Admission location in hospital'));
+
+UPDATE temp_encounter
+SET 
+    transfer_out_location = OBS_VALUE_CODED_LIST(encounter_id,
+            'PIH',
+            'Transfer out location',
+            'en');
+
+UPDATE temp_encounter
+SET 
+    clinical_management_plan = OBS_VALUE_TEXT(encounter_id, 'CIEL', '162749');
+
+UPDATE temp_encounter
+SET 
+    nursing_note = OBS_VALUE_TEXT(encounter_id, 'CIEL', '166021');
+
+UPDATE temp_encounter
+SET 
+    mh_referral = OBS_SINGLE_VALUE_CODED(encounter_id,
+            'CIEL',
+            '1272',
+            'PIH',
+            '5489');
+
+UPDATE temp_encounter 
+SET 
+    mh_note = OBS_VALUE_TEXT(encounter_id, 'CIEL', '161011');
+
+### COVID 19 Progress FORM
+-- overall_condition
+update temp_encounter set overall_condition = obs_value_coded_list(
+	encounter_id,
+    'CIEL',
+    '159640',
+    'fr'
+);
+
+-- new symptom
+UPDATE temp_encounter te LEFT JOIN 
```

Comment:
We're using functions well in all these below, but there must be a way to simplify this use case into a more concise function.